src/decoders/base32_decoder.rs

```diff
@@ -0,0 +1,189 @@
+use crate::checkers::CheckerTypes;
+use crate::decoders::interface::check_string_success;
+
+use super::crack_results::CrackResult;
+///! Decode a base64 string
+///! Performs error handling and returns a string
+///! Call base64_decoder.crack to use. It returns option<String> and check with
+///! `result.is_some()` to see if it returned okay.
+///
+use super::interface::Crack;
+use super::interface::Decoder;
+
+use data_encoding::BASE32;
+use log::{debug, info, trace};
+
+/// The Base64 decoder, call:
+/// `let base64_decoder = Decoder::<Base64Decoder>::new()` to create a new instance
+/// And then call:
+/// `result = base64_decoder.crack(input)` to decode a base64 string
+/// The struct generated by new() comes from interface.rs
+/// ```
+/// use ares::decoders::base64_decoder::{Base64Decoder};
+/// use ares::decoders::interface::{Crack, Decoder};
+/// use ares::checkers::{athena::Athena, CheckerTypes, checker_type::{Check, Checker}};
+///
+/// let decode_base64 = Decoder::<Base64Decoder>::new();
+/// let athena_checker = Checker::<Athena>::new();
+/// let checker = CheckerTypes::CheckAthena(athena_checker);
+///
+/// let result = decode_base64.crack("aGVsbG8gd29ybGQ=", &checker).unencrypted_text;
+/// assert!(result.is_some());
+/// assert_eq!(result.unwrap(), "hello world");
+/// ```
+pub struct Base32Decoder;
+
+impl Crack for Decoder<Base32Decoder> {
+    fn new() -> Decoder<Base32Decoder> {
+        Decoder {
+            name: "base32",
+            description: "Base32 is a group of binary-to-text encoding schemes that represent binary data (more specifically, a sequence of 8-bit bytes) in an ASCII string format by translating the data into a radix-64 representation.",
+            link: "https://en.wikipedia.org/wiki/Base32",
+            tags: vec!["base32", "decoder", "base"],
+            expected_runtime: 0.01,
+            expected_success: 1.0,
+            failure_runtime: 0.01,
+            normalised_entropy: vec![1.0, 10.0],
+            popularity: 1.0,
```

Comment:
I think base62 would probably be 0.7/0.8, 1.0 is for things like base64 / caesar (extremely common to see)

Reply:
I changed it to `0.8`